Userland/Applications/VideoPlayer/main.cpp

```diff
@@ -43,10 +43,12 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
             auto const& frame = block.frame(0);
             dbgln("Reading frame 0 from block @ {}", block.timestamp());
-            bool failed = !vp9_decoder.decode_frame(frame);
+            auto result = vp9_decoder.decode_frame(frame);
             vp9_decoder.dump_frame_info();
-            if (failed)
+            if (result.is_error()) {
+                outln("Error: {}", result.error().string_literal());
```

Comment:
This should be shown to the user since a GUI app might not be run from a terminal
Also errors should maybe be categorized? If the video player was telling me "Padding element was non-zero on boolean decoder exit" I would be a little confused about what this mean. If it's an error because the data is corrupted, make it say "Could not read media, it's likely corrupted. {full_error}"

Reply:
The way I'm planning on using the decoder errors is for it to be a debugging output rather than something the player will display once it's implemented properly. For now, this app is just used to test that the decoder can parse the file correctly, so once it will parse fully I'll have to flesh out the app and create a generic error message with extra information accessible but hidden by default.

If I end up propagating allocation errors or whatever else, I'll want to make another error type for the player to determine whether it was the file at fault or some other issue, but for now all of the errors indicate video corruption, if we're operating under the assumption that the decoder is written correctly (which at the moment it is not).

That being said, after I get past the current parsing roadblock, I'll probably want to create a new error type to store more information about where the errors are occurring in the decoder, since there aren't built in stack traces. I'll easily be able to store an error type enum alongside that.

Reply:
If you prefer that I make the GUI display regardless and put the error on screen, I could certainly do that, but for the moment it won't produce anything meaningful to a user, since it can't read any videos correctly.